### PR TITLE
Capturing support (https://en.cppreference.com/w/cpp/language/lambda)

### PR DESCRIPTION
**Pull request by StephanMeijer**
This enables me to use variables from outside, see:

```

void onConnectionEstablished() {
  client.subscribe("mytopic/test", [](const String & payload) {
    Serial.println(payload);
  });

  for (Light &l : lights) {
    String id = l.getId();

    client.subscribe("light/" + id + "/switch", [&l](const String & payload) {
      if (payload == "ON") {
        client.publish("light/" + l.getId() + "/state", "ON");
      } else if (payload == "OFF") {
        client.publish("light/" + l.getId() + "/state", "OFF");
      }
    });
  }
}
```

### CODE REVIEW

File: src/EspMQTTClient.h [hunk 1]
```diff
@@ -32,7 +32,7 @@
 void onConnectionEstablished(); // MUST be implemented in your sketch. Called once everythings is connected (Wifi, mqtt).
 
 typedef void(*ConnectionEstablishedCallback) ();
-typedef void(*MessageReceivedCallback) (const String &message);
+typedef std::function<void(const String &message)> MessageReceivedCallback;
 typedef void(*MessageReceivedCallbackWithTopic) (const String &topicStr, const String &message);
```
Comment (plapointe6) [inline src/EspMQTTClient.h]: Please apply the std:function changes to this callback as well : 
```c++ 
typedef void(*MessageReceivedCallbackWithTopic) (const String &topicStr, const String &message);
```

File: src/EspMQTTClient.cpp [hunk 2]
```diff
@@ -542,9 +542,9 @@ void EspMQTTClient::mqttMessageReceivedCallback(char* topic, byte* payload, unsi
     if (mqttTopicMatch(mTopicSubscriptionList[i].topic, String(topic)))
     {
       if(mTopicSubscriptionList[i].callback != NULL)
-        (*mTopicSubscriptionList[i].callback)(payloadStr); // Call the callback
+        (mTopicSubscriptionList[i].callback)(payloadStr); // Call the callback
```
Comment (plapointe6) [inline src/EspMQTTClient.cpp]: The parentheses are not required anymore  (mTopicSubscriptionList[i].callback)(payloadStr); ->  mTopicSubscriptionList[i].callback(payloadStr);

**Comment by StephanMeijer** [comment]
I'll do this today or tomorrow

**Comment by plapointe6** [comment]
These changes has been merged with pull request #44

**Pull request closed without merging**